src/db/boxStore.js

```diff
@@ -198,20 +198,26 @@ module.exports.buyIn = async (boxBarcode, boxCount) => {
             .leftJoin('PRICE', 'RVBOX.itembarcode', 'PRICE.barcode')
             .leftJoin('RVITEM', 'PRICE.itemid', 'RVITEM.itemid')
             .where('RVBOX.barcode', boxBarcode)
-            .first('RVBOX.itemcount', 'PRICE.priceid', 'PRICE.count');
+            .first(
+                'RVBOX.itemcount',
+                'PRICE.priceid',
+                'PRICE.count',
+                'PRICE.barcode',
+            );
 
         if (row === undefined) {
             return undefined;
         }
 
-        const { count, itemcount, priceid } = row;
+        const { count, itemcount, priceid, barcode } = row;
 
         const newCount = count + itemcount * boxCount;
 
         await knex('PRICE')
             .transacting(trx)
             .select({ priceid, endtime: null })
-            .update({ count: newCount });
+            .update({ count: newCount })
+            .where('PRICE.barcode', barcode);
```

Comment:
This could have updated the count of some previous price if the select above doesn't do anything.
Should be:
```javascript
.where({ barcode: barcode, endtime: null })
```

Reply:
see previous conversation